cmd/cluster/core/dump.go

```diff
@@ -166,20 +150,67 @@ func dumpGuestCluster(ctx context.Context, opts *DumpOptions) error {
 	}
 	podPort := supportutil.BindAPIPortWithDefaultFromHostedCluster(hostedCluster, config.DefaultAPIServerPort)
 	forwarderStop := make(chan struct{})
-	if err := forwarder.ForwardPorts([]string{fmt.Sprintf("%d", podPort)}, forwarderStop); err != nil {
+	if err := forwarder.ForwardPorts([]string{fmt.Sprintf("%d:%d", localPort, podPort)}, forwarderStop); err != nil {
 		return fmt.Errorf("cannot forward kube apiserver port: %w, output: %s", err, forwarderOutput.String())
 	}
 	defer close(forwarderStop)
 
 	opts.Log.Info("Dumping guestcluster", "target", target)
-	if err := DumpGuestCluster(ctx, opts.Log, kubeconfigFile.Name(), target); err != nil {
+	if err := DumpGuestCluster(ctx, opts.Log, kubeconfigFileName, target); err != nil {
 		return fmt.Errorf("failed to dump guest cluster: %w", err)
 	}
 	opts.Log.Info("Successfully dumped guest cluster", "duration", time.Since(start).String())
 
 	return nil
 }
 
+func createGuestKubeconfig(ctx context.Context, c client.Client, cpNamespace string, localPort int, log logr.Logger) (string, error) {
+
+	localhostKubeconfigSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "localhost-kubeconfig",
+			Namespace: cpNamespace,
+		},
+	}
+	if err := c.Get(ctx, client.ObjectKeyFromObject(localhostKubeconfigSecret), localhostKubeconfigSecret); err != nil {
+		return "", fmt.Errorf("failed to get hostedcluster localhost kubeconfig: %w", err)
+	}
+	kubeconfigFile, err := os.CreateTemp(os.TempDir(), "kubeconfig-")
+	if err != nil {
+		return "", fmt.Errorf("failed to create tempfile for kubeconfig: %w", err)
+	}
+	defer func() {
+		if err := kubeconfigFile.Sync(); err != nil {
+			log.Error(err, "Failed to sync temporary kubeconfig file")
+		}
+		if err := kubeconfigFile.Close(); err != nil {
+			log.Error(err, "Failed to close temporary kubeconfig file")
+		}
+		if err := os.Remove(kubeconfigFile.Name()); err != nil {
```

Comment:
isn't this going to remove the kubeconfig file before `DumpGuestCluster()` can use it?

Reply:
true :(